src/servicesHost.ts

```diff
@@ -509,3 +520,45 @@ function populateDependencyGraphs(
     ] = true;
   });
 }
+
+function addCache(servicesHost: typescript.ModuleResolutionHost) {
+  const clearCacheFuncs: (() => void)[] = [];
+  if (servicesHost.fileExists !== undefined) {
+    const cache = createCache(servicesHost.fileExists);
+    servicesHost.fileExists = cache.cached;
+    clearCacheFuncs.push(cache.clear);
+  }
+
+  if (servicesHost.directoryExists !== undefined) {
+    const cache = createCache(servicesHost.directoryExists);
+    servicesHost.directoryExists = cache.cached;
+    clearCacheFuncs.push(cache.clear);
+  }
+
+  if (servicesHost.realpath !== undefined) {
+    const cache = createCache(servicesHost.realpath);
+    servicesHost.realpath = cache.cached;
+    clearCacheFuncs.push(cache.clear);
+  }
+
+  return () => clearCacheFuncs.forEach(clear => clear());
+}
+
+function createCache<TOut>(func: (arg: string) => TOut) {
+  let cache: Record<string, TOut | undefined> = {};
```

Comment:
I wonder if it might be slightly more performant to use a `Map` instead?  What do you think? One of the big webpack 4 changes was it moving to using `Map`s.

Reply:
Oh, you're right! _I need to start to forget to use simple object as map to be compatible on all "platforms" 🙂_